pep/app/authenticators/ip.ts

```diff
@@ -29,9 +29,17 @@ export default class IpAuthenticator extends CredentialsAuthenticator {
                 const params = serializeQueryParams({ SessionId: sessionData.SessionId });
                 url = `${url}?${params}`;
             }
+            const headers = this.authenticationHeaders;
+            if (this.fastboot.isFastBoot) {
+                const referrer = this.fastboot.request.headers.headers['referer'];
```

Comment:
is `fastboot.request.headers` an actual `Headers` api instance? if so, it might be better to do `this.fastboot.request.headers.get('referer')`